README.md

```diff
@@ -63,7 +65,7 @@ $connector = new oliverlorenz\reactphpmqtt\Connector($loop, $resolver, $version)
 
 $p = $connector->create($config['server'], $config['port'], $config['options']);
 $p->then(function(\React\Stream\Stream $stream) use ($connector) {
-    $stream->on('PUBLISH', function(\oliverlorenz\reactphpmqtt\packet\Publish $message) {
+    $stream->on(Publish::EVENT, function(Publish $message) {
```

Comment:
> One more Idea. Maybe its useful to have a getter in the abstract ControlPacket class which throws a "NotImplementedExecption" if its not overwritten by the inheriting class.

The same as you currently have `getControlPacketType()`?

```
$stream->emit(Publish::getEvent(), function(Publish $message) {
   //...
});
```
I thought about doing that. I definitely think the syntax with a constant is nicer to use.

I agree that it's inconsistent, having some packets that implement the feature, without them being grouped together by a `Emitable` or `Receivable` interface.
I think you can get away with it here as the types of packets that will have events emitted are predefined in `PacketFactory`, and it's absolutely incorrect for anyone to try to implement any additional packet types (which would go against the MQTT spec).